sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -122,6 +122,14 @@ class SparkSession private(
   val sqlContext: SQLContext = new SQLContext(this)
 
   /**
+   * This is the interface through which the user can get [[JavaSparkContext]]
+   *
+   * @since 2.0.0
+   */
+  @transient
+  val javaSparkContext: JavaSparkContext = new JavaSparkContext(sparkContext)
```

Comment:
or we can add this code in spark.javaSparkContext so its more good for user to use it because context is primary thing.